### PR TITLE
fix(benchmarks): compute the chi-squared p-value instead of looking it up

### PR DESCRIPTION
**Pull request by ofri-peretz**
## Summary

- `chiSquaredTest` decided significance with `chiSq > (criticalValues[df] || 5.991)` against a three-row table. Every comparison with `df >= 4` fell through to the **df=2** threshold.
- 5.991 is below the true critical value for **every** `df >= 4` (verified to df=15), so the error had exactly one direction — it manufactured significance and could never withhold a real result. That is why it survived four months: it only ever agreed with what you hoped.
- Replaced with a computed upper-incomplete-gamma tail. No table, no last row to fall off.
- `run-antigravity.js` had the same function with the table extended to five rows and the fallback kept — which moved the cliff from `df>=4` to `df>=6` rather than removing it. Both call sites now use the computed tail.
- `pValue` was the string `"< 0.05"` / `"> 0.05"` rendered from `significant`, restating the boolean and carrying no evidence of its own, while the `df<2` branch put the number `1` on the same field. It is a number on both paths now.

### It fired

`results/ilb-ai/overnight-multi-model-treatment-7iter-2026-02-09.json` recorded `chiSq=18.43, df=4` — inside the gap, judged against 5.991 instead of 9.488. That verdict happened to be **correct anyway** (true p = 0.001017). The right answer from the wrong instrument, which is exactly why nobody noticed.

### Six probes in the gap, all false positives under the old predicate

| χ² | df | old verdict | true p |
|---|---|---|---|
| 7.0 | 4 | significant | 0.1359 |
| 8.0 | 4 | significant | 0.0916 |
| 9.0 | 4 | significant | 0.0611 |
| 7.0 | 5 | significant | 0.2206 |
| 10.0 | 5 | significant | 0.0752 |
| 12.0 | 6 | significant | 0.0620 |

## Test plan

- [x] `npm --prefix benchmarks run stats:check` — 19 assertions, exit 0.
- [x] The check asserts the **ordering rule** the fallback inverted (for a fixed statistic, more df must yield a larger p), so it **fails on the old predicate and passes on the new** — proven before the fix landed, not after.
- [x] Reproduces every textbook α=0.05 critical value to three decimals: 3.841, 5.991, 7.815 — plus 9.488 and 11.07, which the table never had.
- [x] The old predicate and the six probes are committed in `stats.selfcheck.ts`, so the disagreement is re-runnable evidence rather than a throwaway script.
- [x] Bug introduced 2026-05-11 in b40bc6781 (`git log -S criticalValues`).

## After merge

Any consumer reading `pValue` gets a number rather than a rendered verdict. No stored result changes — the one real firing keeps its verdict, it just now has a threshold that justifies it.

Note: `npm ci` currently fails on `main` for unrelated reasons (package.json / package-lock.json out of sync — `Missing: eslint-plugin-crypto@2.2.3`, `eslint-plugin-jsdoc@63.3.3`, `eslint-plugin-jwt@2.2.14`). Worth a separate look.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **New Features**
  - Chi-squared significance tests now calculate precise p-values across supported degrees of freedom.
  - Results include numeric p-values rounded to four significant digits.
  - Significance is determined directly from whether the p-value is below 0.05.

- **Bug Fixes**
  - Improved accuracy for tests with degrees of freedom beyond the previous lookup range.
  - Corrected p-value behavior for varying degrees of freedom, tail probabilities, and infinite statistics.

- **Tests**
  - Added regression checks covering known values, invalid inputs, and edge cases.

### CODE REVIEW

File: benchmarks/lib/stats.selfcheck.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
+/**
+ * Self-check for the significance helpers. Run: npm run -w @interlace/benchmarks stats:check
+ *
+ * The first block is the REGRESSION: ilb-ai tested every run against a
+ * three-entry critical-value table, `criticalValues[df] || 5.991`, so df >= 4
+ * silently borrowed the df=2 threshold. df=4 / chiSq=7.0 is inside that gap —
+ * the old code called it significant, the true tail probability is ~0.136.
+ * Asserts the RULE (a computed tail), not the absence of one bad number.
+ */
+import assert from 'node:assert/strict';
```
[the remark this answers is not in the document]
Comment (ofri-peretz) [reply benchmarks/lib/stats.selfcheck.ts:10]: Fixed in 2b1462972 — took option 2's spirit with option 1's automatic collection: `benchmarks/__tests__/stats-selfcheck-lock.test.ts`.

Kept `stats:check` intact rather than renaming, because the SDLC spec for the companion article cites that exact command as its evidence. The lock imports the module — it asserts at import time, so importing runs all of it — and a second assertion pins that the script still points at this file.

It also asserts the exported count, because an import of a file whose assertions had all been deleted resolves perfectly happily, and a lock that passes when its subject is gone is not a lock. Proved it fails in both directions:

```
revert the Infinity guard  -> FAIL: an infinite statistic is p=0, not NaN
delete every assertion     -> FAIL: expected 0 to be >= 21
restored                   -> 2 passed
```

File: benchmarks/lib/stats.ts [hunk 2]
```diff
@@ -219,3 +219,77 @@ export function accuracyReport(observations, opts = {}) {
...
+ * the df=2 value (5.991). Any run with 5+ groups (df >= 4) was tested against
+ * a threshold below its own, and 5.991 sits under the true critical value for
+ * every df >= 4, so the error only ever manufactured significance. A table has
+ * a silent edge; a computed tail does not.
+ *
+ * @param {number} chiSq  the test statistic (>= 0)
+ * @param {number} df     degrees of freedom (>= 1)
+ * @returns {number} P(X >= chiSq) under the null
+ */
+export function chiSquaredPValue(chiSq, df) {
```
[the remark this answers is not in the document]
Comment (ofri-peretz) [reply benchmarks/lib/stats.ts:306]: You are right, and it is worse than unreachable-in-practice. Reproduced it against the same `upperGamma` with only the guard reverted:

```
OLD chiSquaredPValue(Infinity, 4) => NaN
(NaN < 0.05) => false
```

So the failure mode is not a crash — it is that the most significant statistic there is would read *not significant*, the same direction of error as the lookup table this PR removes. That is worth a line even though the runners only ever produce ratios of integer counts.

Fixed in 2b1462972, close to your second form:

```ts
if (chiSq === Infinity) return df >= 1 ? 0 : 1;
if (!Number.isFinite(chiSq) || chiSq < 0 || !(df >= 1)) return 1;
```

Split in two so the Infinity case reads as an answer (all the mass is in the tail, p = 0) rather than as an input rejection. Covered by a new assertion, plus your df=6 point from the third thread — 12.592 -> p=0.05, the far side of the gap the old table fell into. Self-check is now 21 assertions.
